apps/arm/main.cpp

```diff
@@ -218,41 +218,32 @@ static CANMsg::CANMsgHandlerMap canHandlerMap = {
 
 // Interface and recieve buffer
 CAN can1(CAN1_RX, CAN1_TX, ROVER_CANBUS_FREQUENCY);
-// CANBuffer rxCANBuffer(can1, CANBuffer::BufferType::rx);
+CANBuffer rxCANBuffer(can1, CANBuffer::BufferType::rx);
```

Comment:
what about a tx buffer?

Reply:
Shouldn't really need a tx buffer. ROS should have RX buffer

Reply:
If we send too quickly, wouldn't we be dropping msgs?

Reply:
Then we should send less quickly... there's no point in us sending more than the ROS side can consume especially when in comes to the streamed real-time data. Would you want outdated position messages that were queued because FW didn't get an ack or just miss some? Though maybe we should queue or at least re-try sending the important, 1-time messages like a control-mode ack.

Reply:
It also depends if the ROS side CAN controller sets an ack bit or not when its HW FIFO buffer is full.

Reply:
I dont think it's an issue of the ros side not consuming it fast enough. It's an issue of the fw not sending it fast enough. Correct me if I'm misunderstanding  how it works but:

The hw buffer has 3 msg slots.
If we have more than three can writes within a piece of code, back to back, msgs get overwritten before they get to the point of being sent over the can bus

Reply:
That's true. I'm not really sure how quickly the messages are removed from the HW fifo and it could take longer than it takes to fill it. Esp. if the bus is blocked by higher priority messages.